test/extended/util/docker.go

```diff
@@ -0,0 +1,85 @@
+package util
+
+import (
+	"fmt"
+	dockerClient "github.com/fsouza/go-dockerclient"
+	tutil "github.com/openshift/origin/test/util"
+)
+
+func TagImage(tagee, tagor string) error {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return dcerr
+	}
+	opts := dockerClient.TagImageOptions{
+		Repo:  tagee,
+		Tag:   "latest",
+		Force: true,
+	}
+	terr := client.TagImage(tagor, opts)
+	return terr
+}
+
+func PullImage(name string) error {
+	client, dcerr := tutil.NewDockerClient()
+	if dcerr != nil {
+		return dcerr
+	}
+	opts := dockerClient.PullImageOptions{
+		Repository: name,
+		Tag:        "latest",
+	}
+	perr := client.PullImage(opts, dockerClient.AuthConfiguration{})
+	return perr
+}
+
+type MissingTagError struct {
+	Tags []string
+}
+
+func (mte MissingTagError) Error() string {
+	return fmt.Sprintf("the tag %s passed in was invalid, and not found in the list of images returned from docker", mte.Tags)
+}
+
+func GetImageIDForTags(comps []string) ([]string, error) {
```

Comment:
I have strange feeling that we are doing this in many places ;-) Maybe worth to look around the build/imagestream code if this does not exists already.